ext/yahttp/yahttp/reqresp.hpp

```diff
@@ -249,7 +249,7 @@ namespace YaHTTP {
     }
     void setup(const std::string& method_, const std::string& url_) {
       this->url.parse(url_);
-      this->headers["host"] = this->url.host;
+      this->headers["host"] = this->url.host.find(":") == std::string::npos ? this->url.host : "[" + this->url.host + "]";
```

Comment:
This would break if `:9100` for a port number ever made it into the host header (which is legal).

Reply:
At the moment, YaHTTP doesn't add the port anyway,  that is stored in `url.port`.